### DISCUSSION TITLE
Possible to generate a new handlebars file which includes partials?

### DISCUSSION THREAD

**Discussion by benjick** [handlebars-lang/handlebars.js#1829] [Q&A]
Hello

We're using handlebars for our email templates, one handlebars file per template. Using Mailgun you can't upload partials and reference them and some of our templates share things like footer. So I'm wondering if there's a good way to compile a new handlebars template which bakes in the partials? I'm figuring I can just do a reg-exp replace but I wanted to check if there's an "official" way of doing so.

Cheers

**Comment by benjick** [comment]
Bump

**Reply by jaylinski** [reply]
> I'm wondering if there's a good way to compile a new handlebars template which bakes in the partials

I'm not aware of an easy way to do this, since (as far as I know) the parser doesn't include the partial as a string, but transforms it to an AST.
This would also not work if you are using [dynamic partials](https://github.com/handlebars-lang/handlebars.js/blob/4.x/spec/partials.js#L25-L38), because the included partial is dependent on the data you provide.
